inspirehep/records/api/literature.py

```diff
@@ -44,32 +50,60 @@
 CROSSREF_URL = "https://api.crossref.org/works/<ID>"
 
 
-class LiteratureRecord(InspireRecord, CitationMixin):
+class LiteratureRecord(FilesMixin, CitationMixin, InspireRecord):
     """Literature Record."""
 
     es_serializer = LiteratureESEnhancementV1
     pid_type = "lit"
     pidstore_handler = PidStoreLiterature
 
+    @property
+    def earliest_date(self):
+        """Returns earliest date. If earliest date is missing month or day
+        it's set as 1 as DB does not accept date without day or month
+
+        Returns:
+            str: earliest date represented in a string
+        """
+        date_paths = [
+            "preprint_date",
+            "thesis_info.date",
+            "thesis_info.defense_date",
+            "publication_info.year",
+            "legacy_creation_date",
+            "imprints.date",
+        ]
+
+        dates = [
+            str(el)
+            for el in chain.from_iterable(
+                force_list(self.get_value(path)) for path in date_paths
+            )
+        ]
+        proper_dates = []
+        for date in dates:
+            try:
+                proper_dates.append(dateutil.parser.parse(date))
+            except ValueError:
+                pass
+        if proper_dates:
+            date = earliest_date(dates, full_date=True)
+            return date
+        return None
+
     @classmethod
     def create(cls, data, **kwargs):
-        documents = data.pop("documents", None)
-        figures = data.pop("figures", None)
-        record = super().create(data, **kwargs)
-        if documents or figures:
-            record.set_files(documents=documents, figures=figures)
```

Comment:
Why you are removing this like that? Is it not easier to disable flag in config?

Reply:
it's completely wrong, especially on update it doesn't take the new ``data`` it takes ``self`` we have to fix this for literature

Reply:
Thanks for explaining this IRL.